### ISSUE TITLE
PSer, PhSer processed as series

### ISSUE THREAD

**Issue by BjornEnemark** [thomas-maeder/popeye#337]
Ser-H#, Ser-# and Ser-S# are solved very fast. I think that Popeye keeps a table of positions previously obtained, so that sequences like 1. Se4 2. Sf6 3. Se4 is not examined further. The same principle should be used for PSer-/PhSer- problems

**Comment by JoshuaGreen** [comment]
I doubt that dramatic improvement here will be so simple, but just to pin down the request, are you thinking of ordinary solving or intelligent mode solving?

**Comment by BjornEnemark** [comment]
I am thinking of ordinary solving for _selfmate_ seriesmovers (Ser-S# or 
Pser-S#), no intelligent mode possible as far as I know.
If a position is reached that has been encountered before (like 1. Sf6 
2. Sg4 3. Sf6) then this branch should be cut.
As I understand it, solving of seriesmovers for helpmates will do just 
that: check each move to see if the current position has been seen 
before in the solution. All intermediate positions are stored in memory, 
controlled by maxmem.
I think that Pser-H# are not treated like this in ordinary solving.

Bjørn


Den 28-05-2022 kl. 02:47 skrev Joshua Green:

**Comment by JoshuaGreen** [comment]
Yeah, intelligent mode doesn't work for selfmates (though I feel that with some work it could be effective against reflexmates, Issue #315).  At any rate, though my understanding of the infrastructure is rather limited, I'm sure that positions _are_ cached in `pser` problems.  The issue is that any piece can potentially move, hence a lot more positions may in principle be stored, and that can require a lot more memory (both in total, and possibly per position).
